### ISSUE TITLE
asyncronous evaluate, aka done callback

### ISSUE THREAD

**Issue by daraosn** [segmentio/nightmare#648]
Is there a way to asynchronously `.evaluate()`?

I was taking a look at [horseman](https://github.com/johntitus/node-horseman), and they provide a handy `done` callback...
```
horseman
  .open('http://en.wikipedia.org/wiki/Headless_Horseman')
  .evaluate(function(ms, done){
    var start = Date.now();
    setTimeout(function() {
      done(null, Date.now() - start);
   ..
```

I'm thinking in implementing it myself to nightmare, but want to be sure this doesn't exist already?

**Comment by rosshinkley** [comment]
This is touched on in #552, and a fix is proposed in #573.  At the time of this comment, it's incomplete.

**Comment by rosshinkley** [comment]
Merging into #552.
